pkg/core/pipeline/condition/main.go

```diff
@@ -134,6 +134,19 @@ func (c *Config) Validate() error {
 		}
 	}
 
+	// Handle depends_on deprecation
+	if len(c.DeprecatedDependsOn) > 0 {
+		switch len(c.DependsOn) == 0 {
+		case true:
+			logrus.Warningf("%q is deprecated in favor of %q.", "depends_on", "dependson")
```

Comment:
Why the interpreted string here? I would have used a simple string, it won't change.
(Same below and in the other files)

Reply:
It's more a matter of taste, I prefer the readability this way than something like 

`logrus.Warningln("\"depends_on\" is deprecated in favor of \"dependson\".")`